### PR TITLE
D1 admin action

### PR DESCRIPTION
**Pull request by shubaivan**
form for create project

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,4 +15,6 @@
 /composer.phar
 /vendor/
 /web/bundles/
-/web-src/less/*.css
\ No newline at end of file
+/web-src/less/*.css
+*~
```
Comment (ahonymous) [inline .gitignore:19]: It's should be in the global git ignore

File: .gitignore [hunk 2]
```diff
@@ -15,4 +15,6 @@
...
+.idea
```
Comment (ahonymous) [inline .gitignore:20]: It's too in the global gitignore

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ *
+ * @Route("/admin/projects")
+ */
+class AdminProjectController extends Controller
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: I think than this class name should be as ProjectController

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+{
+    /**
+     * @Route("/show", name="admin_projects_show")
+     * @Template()
+     * @Method({"GET"})
+     */
+    public function listAction()
+    {
+        $security = $this->get('security.context');
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: security context is depricated - never use

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        if ($security->isGranted('ROLE_ADMIN')) {
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: It's already was wrote [this](https://github.com/civils-council/civilsbudget/blob/api/app/config/security.yml#L60)

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    /**
+     * @Route("/{id}", name="admin_project_show", requirements={"id" = "\d+"})
+     * @Template()
+     * @Method({"GET", "POST"})
+     */
+    public function showProjectAction(Project $project, Request $request)
+    {
+        $security = $this->get('security.context');
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: remove

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        if ($security->isGranted('ROLE_ADMIN')) {
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: remove

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    /**
+     * @Route("/add", name="admin_projects_add")
+     * @Method({"GET", "POST"})
+     * @Template()
+     */
+    public function addProjectAction(Request $request)
+    {
+        $security = $this->get('security.context');
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: remove

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+        if ($security->isGranted('ROLE_ADMIN')) {
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: remove

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+            $entity = new Project();
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: set name variable $entity to $project

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+        // --------------------------------- Create Forms ---------------------------------
+
+        /**
+         * Creates a form to create a Project entity.
+         *
+         * @param Project $entity The entity
+         *
+         * @return \Symfony\Component\Form\Form The form
+         */
+        private function createCreateForm(Project $entity)
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: How much you use this code in the he methods?

File: src/AppBundle/Controller/Admin/AdminProjectController.php [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+    {
+        $form = $this->createForm(new AddProjectType(), $entity, array(
+            'action' => $this->generateUrl('admin_projects_add', array('id' => $entity->getId())),
+            'method' => 'POST',
+            'attr' => array('class' => 'formCreateClass'),
+        ));
+        $this->addFormFields($form);
+        $form->add('submit', 'submit', array('label' => 'Add Project'));
```
Comment (ahonymous) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: [This](https://github.com/civils-council/civilsbudget/blob/api/app/Resources/views/form.html.twig) automatic added submit to form

File: src/AppBundle/Controller/ProjectController.php [hunk 13]
```diff
@@ -62,4 +64,59 @@ public function likeAction(Project $project, Request $request)
...
     }
+
+    /**
+     * @Route("/projects/add", name="projects_add")
+     * @Method({"GET", "POST"})
+     * @Template()
+     */
+    public function addProjectAction(Request $request)
+    {
+        $security = $this->get('security.context');
```
Comment (ahonymous) [inline src/AppBundle/Controller/ProjectController.php]: remove

File: src/AppBundle/Controller/ProjectController.php [hunk 14]
```diff
@@ -62,4 +64,59 @@ public function likeAction(Project $project, Request $request)
...
+        if ($security->isGranted('ROLE_USER')) {
```
Comment (ahonymous) [inline src/AppBundle/Controller/ProjectController.php]: remove

File: src/AppBundle/Form/AddProjectType.php [hunk 15]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace AppBundle\Form;
+
+use Symfony\Component\Form\AbstractType,
+    Symfony\Component\Form\FormBuilderInterface,
+    Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+class AddProjectType extends AbstractType
```
Comment (ahonymous) [inline src/AppBundle/Form/AddProjectType.php]: Meybe name should be ProjectType

File: src/AppBundle/Form/AddProjectType.php [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    /**
+     * @param OptionsResolverInterface $resolver
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        $resolver->setDefaults(array(
+            'data_class' => 'AppBundle\Entity\Project',
+            'csrf_protection' => false
```
Comment (ahonymous) [inline src/AppBundle/Form/AddProjectType.php]: remove

File: src/AppBundle/Form/AddProjectType.php [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+        ));
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'add_project';
```
Comment (ahonymous) [inline src/AppBundle/Form/AddProjectType.php]: return "project";

File: .gitignore (hunk 1 above)
Comment (shubaivan) [inline .gitignore:19]: ok, but, some developer have not global git ignore

File: src/AppBundle/Controller/Admin/AdminProjectController.php (hunk 4 above)
Comment (shubaivan) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: yes. but I try with debug and security.context work

File: src/AppBundle/Controller/Admin/AdminProjectController.php (hunk 5 above)
Comment (shubaivan) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: I know, this is double security. I delete this 'if'

File: src/AppBundle/Controller/Admin/AdminProjectController.php (hunk 6 above)
Comment (shubaivan) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: ok

File: src/AppBundle/Controller/Admin/AdminProjectController.php (hunk 11 above)
Comment (shubaivan) [inline src/AppBundle/Controller/Admin/AdminProjectController.php]: This advice from right way

File: src/AppBundle/Controller/ProjectController.php (hunk 13 above)
Comment (shubaivan) [inline src/AppBundle/Controller/ProjectController.php]: done

File: src/AppBundle/Form/AddProjectType.php (hunk 15 above)
Comment (shubaivan) [inline src/AppBundle/Form/AddProjectType.php]: done

File: src/AppBundle/Form/AddProjectType.php (hunk 16 above)
Comment (shubaivan) [inline src/AppBundle/Form/AddProjectType.php]: setDefaultOptions?

File: src/AppBundle/Form/AddProjectType.php (hunk 17 above)
Comment (shubaivan) [inline src/AppBundle/Form/AddProjectType.php]: done

**Pull request merged**